highfive MPIOFileDriver deprecation

```
/path/to/phare/src/hdf5/detail/h5/h5_file.hpp: In static member function ‘static auto PHARE::hdf5::h5::HighFiveFile::createHighFiveFile(std::string, unsigned int, bool)’:
/path/to/phare/src/hdf5/detail/h5/h5_file.hpp:71:75: warning: ‘MPIOFileDriver’ is deprecated: "Add MPIOFileAccess directly to FileAccessProps" [-Wdeprecated-declarations]
   71 |                     HighFive::MPIOFileDriver(MPI_COMM_WORLD, MPI_INFO_NULL)
      |                                                                           ^
In file included from /path/to/phare/subprojects/highfive/include/highfive/H5File.hpp:14,
                 from /path/to/phare/src/hdf5/detail/h5/h5_file.hpp:4,
                 from /path/to/phare/src/diagnostic/detail/h5writer.hpp:11,
                 from /path/to/phare/src/diagnostic/diagnostics.hpp:27,
                 from /path/to/phare/src/simulator/phare_types.hpp:7,
                 from /path/to/phare/src/simulator/simulator.hpp:8,
                 from /path/to/phare/src/python3/cpp_etc.cpp:3:
/subprojects/highfive/include/highfive/H5FileDriver.hpp:27:71: note: declared here
   27 | class H5_DEPRECATED("Add MPIOFileAccess directly to FileAccessProps") MPIOFileDriver
```

this has been upgraded to catastrophic error

```
/home/runner/work/PHARE/PHARE/src/hdf5/detail/h5/h5_file.hpp:71:31: error: ‘MPIOFileDriver’ is not a member of ‘HighFive’; did you mean ‘FileDriver’?
   71 |                     HighFive::MPIOFileDriver(MPI_COMM_WORLD, MPI_INFO_NULL)
```

--- a/res/cmake/dep/highfive.cmake
+++ b/res/cmake/dep/highfive.cmake
@@ -10,16 +10,10 @@ if(HighFive)
       COMMAND ${Git} clone https://github.com/BlueBrain/HighFive ${HIGHFIVE_SRC} -b master --depth 1
     )
 
-    # Remove symlinks that cause infinite looping
-    #  https://github.com/BlueBrain/HighFive/issues/338
-    file(REMOVE_RECURSE ${HIGHFIVE_SRC}/tests)
-
   else()
     if(devMode)
       message("downloading latest HighFive updates")
-      execute_process(COMMAND ${Git} checkout tests WORKING_DIRECTORY ${HIGHFIVE_SRC}) # undo tests delete
       execute_process(COMMAND ${Git} pull origin master WORKING_DIRECTORY ${HIGHFIVE_SRC})
-      file(REMOVE_RECURSE ${HIGHFIVE_SRC}/tests) # redelete symlinks
     endif(devMode)
   endif()
 

--- a/src/hdf5/detail/h5/h5_file.hpp
+++ b/src/hdf5/detail/h5/h5_file.hpp
@@ -68,6 +68,8 @@ public:
         {
 #if defined(H5_HAVE_PARALLEL)
             fapl.add(HighFive::MPIOFileAccess{MPI_COMM_WORLD, MPI_INFO_NULL});
+#else
+            std::cout << "WARNING: PARALLEL HDF5 not available" << std::endl;
 #endif
         }
         return HiFile{path, flags, fapl};

--- a/src/hdf5/detail/h5/h5_file.hpp
+++ b/src/hdf5/detail/h5/h5_file.hpp
@@ -60,22 +60,22 @@ auto vector_for_dim()
 class HighFiveFile
 {
 public:
-    static auto createHighFiveFile(std::string const path, unsigned flags, bool para)
+    template<typename FileAccessProps>
+    static auto createHighFiveFile(std::string const path, unsigned flags, bool para,
+                                   FileAccessProps& fapl)
     {
         if (para)
-            return HiFile
-            {
-                path, flags
+        {
 #if defined(H5_HAVE_PARALLEL)
-                    ,
-                    HighFive::MPIOFileDriver(MPI_COMM_WORLD, MPI_INFO_NULL)
+            fapl.add(HighFive::MPIOFileAccess{MPI_COMM_WORLD, MPI_INFO_NULL});
 #endif
-            };
-        return HiFile{path, flags};
+        }
+        return HiFile{path, flags, fapl};
     }
 
     HighFiveFile(std::string const path, unsigned flags = HiFile::ReadWrite, bool para = true)
-        : h5file_{createHighFiveFile(path, flags, para)}
+        : fapl_{}
+        , h5file_{createHighFiveFile(path, flags, para, fapl_)}
     {
     }
 
@@ -222,6 +222,7 @@ public:
     HighFiveFile& operator=(const HighFiveFile&&) = delete;
 
 private:
+    HighFive::FileAccessProps fapl_;
     HiFile h5file_;
 
 

--- a/tests/core/utilities/indexer/test_indexer.cpp
+++ b/tests/core/utilities/indexer/test_indexer.cpp
@@ -278,7 +278,7 @@ TEST(Indexer, endReturnsIteratorOnLastPlusOneElement)
     for (std::size_t itemIndex = 0; itemIndex < values.size(); ++itemIndex)
         indexer.add(itemIndex);
 
-    int actual;
+    int actual = 0;
     for (auto const& itemIndex : indexer)
     {
         actual = values[itemIndex];
@@ -293,7 +293,7 @@ TEST(Indexer, singleElementIndexerHasCorrectEnd)
     for (std::size_t itemIndex = 0; itemIndex < values.size(); ++itemIndex)
         indexer.add(itemIndex);
 
-    int actual;
+    int actual = 0;
     for (auto const& itemIndex : indexer)
     {
         actual = values[itemIndex];
@@ -405,4 +405,3 @@ int main(int argc, char** argv)
 
     return RUN_ALL_TESTS();
 }
-

